pkg/state/state_cni_plugins.go

```diff
@@ -33,29 +33,31 @@ import (
 	"github.com/Mellanox/network-operator/pkg/config"
 	"github.com/Mellanox/network-operator/pkg/consts"
 	"github.com/Mellanox/network-operator/pkg/render"
-	"github.com/Mellanox/network-operator/pkg/staticconfig"
 	"github.com/Mellanox/network-operator/pkg/utils"
 )
 
 const stateCNIPluginsName = "state-container-networking-plugins"
 const stateCNIPluginsDescription = "Container Networking CNI Plugins deployed in the cluster"
 
 // NewStateCNIPlugins creates a new state for secondary container networking CNI plugins
-func NewStateCNIPlugins(k8sAPIClient client.Client, scheme *runtime.Scheme, manifestDir string) (State, error) {
+func NewStateCNIPlugins(
```

Comment:
we can embed MainfestRenderer into State and avoid these changes (returning same obj ptr twice)

then relevant code can just accept MainfestRenderer and not whole State obj.

WDYT ?

Reply:
Yeah I think in this case the code will be easier to extend

Reply:
Looking into it, we can't really do that as the interface requires NicClusterPolicy object to be passed into the GetManifestObjects func. I think it's not worth adding such a method to the states not related to NicClusterPolicy